app/src/main/kotlin/compiler/intermediate/Register.kt

```diff
@@ -1,23 +1,25 @@
 package compiler.intermediate
 
 // A class representing virtual cpu register
-class Register {
+class Register(private val asmName: String = "VIRTUAL REGISTER") {
+    override fun toString() = asmName
```

Comment:
It would be more readable to use a dedicated function like `toAsm` instead of the generic `toString`.

Reply:
Done